drawBot/context/baseContext.py

```diff
@@ -403,14 +412,30 @@ def _getCGPath(self):
             elif instruction == AppKit.NSClosePathBezierPathElement:
                 Quartz.CGPathCloseSubpath(path)
         # hacking to get a proper close path at the end of the path
-        x, y, _, _ = self.bounds()
-        Quartz.CGPathMoveToPoint(path, None, x, y)
-        Quartz.CGPathAddLineToPoint(path, None, x, y)
-        Quartz.CGPathAddLineToPoint(path, None, x, y)
-        Quartz.CGPathAddLineToPoint(path, None, x, y)
-        Quartz.CGPathCloseSubpath(path)
+        if withBounds:
+            x, y, _, _ = self.bounds()
+            Quartz.CGPathMoveToPoint(path, None, x, y)
+            Quartz.CGPathAddLineToPoint(path, None, x, y)
+            Quartz.CGPathAddLineToPoint(path, None, x, y)
+            Quartz.CGPathAddLineToPoint(path, None, x, y)
+            Quartz.CGPathCloseSubpath(path)
         return path
 
+    def _setCGpath(self, cgpath):
```

Comment:
the camelcase method name should be `_setCGPath`

Reply:
fixed